### PR TITLE
feat(save): add --drop flag to remove dataset components when saving

### PR DESCRIPTION
**Pull request by b5**
I'm not sure this should be merged as-is (which is why I've skipped tests for the moment).

We're in a situation where a few versions of qri have been creating bad DAG structures, referencing blocks not stored within the created DAG. Thanks to save being a patch operation, and FSI-linked save not recognizing the viz component, qri presents no way to remove a bad viz component. This --drop flag at least lets us drop entire component for now upon save.

Example uses:
```
# drop viz
$ qri save --drop viz

# add stuff, and drop readme
$ qri save --file body.csv --drop rm
```

### CODE REVIEW

File: base/save.go [hunk 1]
```diff
@@ -240,3 +246,43 @@ func GenerateAvailableName(r repo.Repo, peername, prefix string) string {
...
+		case "vz":
+			ds.Viz = nil
+		case "tf":
+			ds.Transform = nil
+		case "st":
+			ds.Structure = nil
+		case "bd":
+			ds.Body = nil
+		case "rd":
+			if ds.Viz != nil {
```
Comment (dustmop) [inline base/save.go:277]: Could you add a comment on why Viz is handled differently here.

File: lib/datasets.go [hunk 2]
```diff
@@ -430,6 +430,8 @@ type SaveParams struct {
 	ConvertFormatToPrev bool
 	// string of references to recall before saving
 	Recall string
+	// string of references to delete before saving
```
Comment (dustmop) [inline lib/datasets.go:433]: Not "references". Specify that it's a comma separated list of component names.

File: base/save.go (hunk 1 above)
Comment (b5) [reply base/save.go:277]: "rd" is a stand in for "rendered", which isn't a component. I'm going to remove support for dropping `rd` for now, as it's not a component

File: base/save.go [hunk 3]
```diff
@@ -240,3 +246,39 @@ func GenerateAvailableName(r repo.Repo, peername, prefix string) string {
...
+	revs, err := dsref.ParseRevs(revStr)
+	if err != nil {
+		return err
+	}
+
+	for _, rev := range revs {
+		if rev.Gen != 1 {
+			return fmt.Errorf("cannot drop specific generations")
+		}
+		switch rev.Field {
```
Comment (Arqu) [inline base/save.go:265]: Could we have this live in `base/component` like https://github.com/qri-io/qri/blob/ef62c71198c2110b5d4de5c5cf7245c78d8a9aed/base/component/field_id.go
And tag these for this specific usecase.

Comment (b5) [reply base/save.go]: don't think it should move into the component package. We haven't fully documented this, but the component package handles dynamic behavior on components using a new internal API. If anything those values `field_id.go` should move into the `dsref` package.

That said, I'm very much with you, `Drop` shouldn't be in `save.go`, so I've moved it to `base/revisions.go` and added tests.

**Pull request merged**